S3Proxy/cache.go

```diff
@@ -0,0 +1,49 @@
+package S3Proxy
+
+import (
+	"encoding/json"
+	"time"
+)
+
+var s3Buckets = map[string]*s3BucketCacheItem{}
+
+type s3BucketCacheItem struct {
+	Name      string
+	Location  string
+	Timestamp time.Time
+	TTL       time.Duration
+}
+
+func (b s3BucketCacheItem) String() string {
+	tmp := &b
+	tmp.TTL = b.TTL - time.Since(b.Timestamp)
+	out, _ := json.Marshal(tmp)
+	return string(out)
```

Comment:
Jip this is correct. You are passing by value now, so the original bucket you passed does not get modified.